akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -104,19 +104,27 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
   private def snapshotFile(metadata: SnapshotMetadata, extension: String = ""): File =
     new File(snapshotDir, s"snapshot-${URLEncoder.encode(metadata.persistenceId, "UTF-8")}-${metadata.sequenceNr}-${metadata.timestamp}${extension}")
 
-  private def snapshotMetadata(persistenceId: String, criteria: SnapshotSelectionCriteria): immutable.Seq[SnapshotMetadata] =
-    snapshotDir.listFiles(new SnapshotFilenameFilter(persistenceId)).map(_.getName).collect {
+  private def snapshotMetadata(persistenceId: String, criteria: SnapshotSelectionCriteria): immutable.Seq[SnapshotMetadata] = {
+    val files = snapshotDir.listFiles(new SnapshotFilenameFilter(persistenceId))
+    if (files eq null) Nil // if the dir was removed
+    else files.map(_.getName).collect {
       case FilenamePattern(pid, snr, tms) ⇒ SnapshotMetadata(URLDecoder.decode(pid, "UTF-8"), snr.toLong, tms.toLong)
     }.filter(md ⇒ criteria.matches(md) && !saving.contains(md)).toVector
+  }
 
   override def preStart() {
-    if (!snapshotDir.isDirectory) {
+    snapshotDir()
+    super.preStart()
+  }
+
+  private def snapshotDir(): File = {
+    if (!dir.isDirectory) {
```

Comment:
a better way i think would be
if(dir.exists){
if it's file
   delete it and then create as dir
}else create as dir